### ISSUE TITLE
The selected default location is not remembered

### ISSUE THREAD

**Issue by BharatKalluri** [lainsce/yishu#26]
Hi, 

If I select default location as Dropbox and close the application and reopen it. The app still tries to get todo.txt from home folder.

### Some Investigation

- Even though I look in the dconf editor and all is well there, it looks like yishu is overwriting dconf entries every time the app is opened. I assume this is only happening to `todo_txt_file_path`.

I think if that is solved, all will be good. I tried tracing back the issue in the code but did not succeed. Looking forward to your fix, so that I can learn from it.

**Comment by lainsce** [comment]
I'll see what I can do

**Comment by lainsce** [comment]
Fixed in `master`.

**Comment by BharatKalluri** [comment]
Hey @lainsce ,

I updated the app from the app center and the problem still persists.
![yishu](https://user-images.githubusercontent.com/8049414/47378213-161bb600-d715-11e8-89f0-86a569c90bbc.gif)

**Comment by lainsce** [comment]
Try resetting all of the keys in dconf, I think a fresh gsettings will help you:
`dconf reset -f /com/github/lainsce/yishu/`
If it still persists after that, I really don't know how to fix this, might be a bigger bug than expected.

**Comment by BharatKalluri** [comment]
Hey @lainsce ,
I tried resetting keys in dconf, but the issue still persists.

**Comment by HerrAugust** [comment]
Same issue for me, Elementary OS 5, Yishu 1.1.6.

Start Yishu, Preferences, select a Custom Location, restart the app. The location is not remembered

**Comment by lainsce** [comment]
Yeah, I don`t know why it happens for some people and not others. Still, it is closed as the code is made and it should always save.

**Comment by dar5hak** [comment]
I encountered this very same issue so I decided to look at the code.

### TL;DR

When a custom location is set, how should the **Default Location** field behave?

### Long story

It was being reset on [this line](https://github.com/lainsce/yishu/blob/master/src/Application.vala#L126) in _Application.vala_. Removing that line made sure the GSetting is no longer reset every time the app starts.

However, as soon as you open **Preferences**, it was reset again. That was due to [this line](https://github.com/lainsce/yishu/blob/master/src/Preferences.vala#L58) in _Preferences.vala_. I removed that as well.

The only problem now is that the **Default Location** has no value set because [this else-if ladder](https://github.com/lainsce/yishu/blob/master/src/Preferences.vala#L61-L76) doesn't match any branch.

What should go into the `else` here?

### Talk is cheap

Here's my [branch](https://github.com/lainsce/yishu/compare/master...elfenware:custom-file).

**Comment by dar5hak** [comment]
Also, I have a feeling those lines were there for good reason and me simply lynching them probably causes problems I cannot foresee.

**Comment by lainsce** [comment]
@dar5hak Can you make that into a Pull Request? We can discuss further this in there.
